packages/styles/src/components/forms.scss

```diff
@@ -51,25 +51,90 @@ select.form-control-rg:not([size]):not([multiple]) {
       border-left: inherit;
     }
 
+    // Fix Safari cutoff when change the initial position of the selector button.
+    @supports (background: -webkit-named-image(i)) {
+      &::file-selector-button {
+        display: initial;
+        position: initial;
+        margin-inline: 0;
+        padding-inline: 0;
+        width: 0;
+        border: 0 none;
+      }
+
+      &::after {
+        display: flex;
+        align-items: center;
+        content: 'Choose File';
+        position: absolute;
+        top: 0;
+        bottom: 0;
+        right: 0;
+        z-index: 2;
+        padding-inline: forms.$input-padding-y;
```

Comment:
This is needed for `valid` and `invalid` state. Previously, we inherited padding value from Bootstrap. It works for all cases except those valid and invalid state where the padding right and left aren't the same. Issue was already there in https://github.com/swisspost/design-system/issues/1297